### PR TITLE
Add easy AI to TicTacToe

### PR DESCRIPTION
**Pull request by jasonharrison**
Add easy AI to TicTacToe
---
The current AI for TicTacToe uses BGIO's MCTS, and it does not let users win.

This PR adds an easy "AI" which makes moves randomly. 

The easy AI is now the default, and the MCTS AI remains as the "hard" option.

A dropdown for selecting the difficulty was also added as IGameModeExtraInfoDropdown.

### CODE REVIEW

File: src/App/Game/GameModePicker.tsx [hunk 1]
```diff
@@ -144,6 +152,34 @@ export class GameModePicker extends React.Component<IGameModePickerProps, IGameM
             />
           </div>
         );
+      } else if (extraInfo.type === 'dropdown') {
+        const dropdown = (extraInfo as IGameModeExtraInfoDropdown);
+        const list: JSX.Element[] = [];
+        dropdown.options.map((option, idx) => {
```
Comment (flamecoals) [inline src/App/Game/GameModePicker.tsx]: Map converts each element of a list into another element of a new list. So you should not use a side effect such push inside map.

Do list = dropdown.options.map yada yada yada return <MenuItem

File: src/games/tictactoe/ai.ts [hunk 2]
```diff
@@ -6,19 +6,45 @@ interface IPlayState {
   ctx: any;
 }
 
+class TictactoeRandomBot {
+  async play(state: IPlayState, playerID: string) {
+    const cell = this.generateMove(state);
+    return this.makeMove(cell);
+  }
+  generateMove(state: IPlayState) {
```
Comment (flamecoals) [inline src/games/tictactoe/ai.ts]: Would be good to say it is random in the name

File: src/App/Game/GameModePicker.tsx (hunk 1 above)
Comment (jasonharrison) [reply src/App/Game/GameModePicker.tsx]: Good idea, thanks!

File: src/games/tictactoe/ai.ts (hunk 2 above)
Comment (jasonharrison) [reply src/games/tictactoe/ai.ts]: I actually had that and then I decided to name the class "TictactoeRandomBot".  But I agree, I'll rename the method.

File: src/App/Game/GameModePicker.tsx [hunk 3]
```diff
@@ -51,7 +58,8 @@ export class GameModePicker extends React.Component<IGameModePickerProps, IGameM
     return React.createElement(Link, { ...props, to }, props.children);
   }
 
-  _handleExtraInfoChange = (mode: GameMode) => (event: any, value: number) => {
+  _handleExtraInfoChange = (mode: GameMode, argValue?: any) => (event: any, value?: number) => {
+    value = value || argValue;
```
[the remark this answers is not in the document]
Comment (jasonharrison) [reply src/App/Game/GameModePicker.tsx]: So the ```<Slider>``` has **onChange**={this._handleExtraInfoChange...}, but my ```<MenuItem>``` has **onClick**={this._handleExtraInfoChange...}.

I found that onClick does not pass a value to _handleExtraInfoChange:

```typescript
TS2322: Type '(event: any, value: number) => void' is not assignable to type '(event: MouseEvent<HTMLElement, MouseEvent>) => void'.
```

I got around this by passing my value as an argument (**argValue**).

[the remark this answers is not in the document]
Comment (flamecoals) [reply src/App/Game/GameModePicker.tsx]: this is also the part that is decreasing the coverage.
I would say, leave _handleExtraInfoChange, maybe rename it to _handleSliderChange, and create a new one _handleClickSelection, and make sure to test a click in this.

File: src/App/Game/GameModePicker.tsx [hunk 4]
```diff
@@ -144,6 +152,33 @@ export class GameModePicker extends React.Component<IGameModePickerProps, IGameM
...
+            <MenuList
+              style={{
+                paddingRight: '20%',
+                display: 'flex',
+                listStyle: 'none',
+              }}
+            >
+              {list}
+            </MenuList>
+          </div>
```
Comment (flamecoals) [inline src/App/Game/GameModePicker.tsx]: this function is too big, can you break it down into smaller ones ? ie getDropDown and getSlider ?

**Pull request merged**